### PR TITLE
qt 5.15 compatibility

### PR DESCRIPTION
**Pull request by ebbit1q**
- fixes deprecation warnings and moves of enums in qt 5.15
- removed old, unused qsrand call
- refactored logic for changing user image 

fixes: #4028

### CODE REVIEW

File: servatrice/src/server_logger.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void ServerLogger::logMessage(QString message, void *caller)
     // filter out all log entries based on values in configuration file
     bool shouldWeWriteLog = settingsCache->value("server/writelog", 1).toBool();
     QString logFilters = settingsCache->value("server/logfilters").toString();
-    QStringList listlogFilters = logFilters.split(",", QString::SkipEmptyParts);
+    QStringList listlogFilters = logFilters.split(",", Qt::SkipEmptyParts);
```
Comment (ctrlaltca) [inline servatrice/src/server_logger.cpp:60]: Qt::SkipEmptyParts was introduced in Qt 5.14, I'm afraid we'll need an #ifdef here

Comment (ebbit1q) [reply servatrice/src/server_logger.cpp:60]: I just noticed a lot more warnings as well :(

**Comment by ebbit1q** [comment]
I fixed all of the easy ones, there are like 3 left that require some more consideration.

File: cockatrice/src/filtertreemodel.cpp [hunk 2]
```diff
@@ -4,6 +4,7 @@
 #include "filtertree.h"
 
 #include <QFont>
+#include <qnamespace.h>
```
Comment (ctrlaltca) [inline cockatrice/src/filtertreemodel.cpp:7]: That looks a private header, maybe `#include <Qt>` will work?

Comment (ebbit1q) [reply cockatrice/src/filtertreemodel.cpp:7]: ah, yeah, that seems to have been autogenerated.

**Comment by ebbit1q** [comment]
I removed the qsrand call, I'm pretty certain that is no longer needed since we use sfmt.

**Comment by ebbit1q** [comment]
Also the refactored logic for changing user image should get some more testing.

**Comment by ctrlaltca** [comment]
> I removed the qsrand call, I'm pretty certain that is no longer needed since we use sfmt.

I agree, qsrand() is used to seed the generator used by qrand(), but I can't find any usage of qrand() in the code

Btw, there are still some `#include <qnamespace.h>` in the patch

**Comment by ebbit1q** [comment]
I err, don't know how those got there.

**Comment by ebbit1q** [comment]
passed but doesn't update >.> https://travis-ci.org/github/Cockatrice/Cockatrice/builds/699346129

**Pull request merged**